### PR TITLE
database for order and order statues done & fix errors

### PR DESCRIPTION
**Pull request by sajalahalih**
add database for each order and order statues and resolve the conflict

### CODE REVIEW

**Review by ayanourr — APPROVED** [review]

File: app/schemas/order_status_schema.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: str
+
+class OrderStatusResponse(BaseModel):
+    id: UUID
+    name: str
+    created_at: datetime
+    updated_at: datetime
+
+    class Config:
+        orm_mode = True
```
Comment (ayanourr) [inline app/schemas/order_status_schema.py]: add( from_attributes=True ) to the config

File: app/api/routes/order_status.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from app.schemas.order_status_schema import OrderStatusCreate, OrderStatusResponse, OrderStatusUpdate
+from app.services.order_status_service import OrderStatusService
+from app.models import User
+
+router = APIRouter()
+
+
+@router.post("/statuses/", response_model=OrderStatusResponse, status_code=status.HTTP_201_CREATED)
+async def create_status(order_status: OrderStatusCreate, current_admin: User = Depends(get_current_admin)):
+    async with Session(engine) as session:
```
Comment (Mohammed-samaheen) [inline app/api/routes/order_status.py]: Please follow the same convention across the entire repo. You were using `session: Session = Depends(get_session)`, so continue using it consistently.

File: app/api/routes/order_status.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return updated_status
+
+
+
+
+@router.delete("/statuses/{status_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def remove_status(status_id: UUID, current_admin: User = Depends(get_current_admin)):
+    async with Session(engine) as session:
+        OrderStatusService.remove_status(session, status_id)
+    return
```
Comment (Mohammed-samaheen) [inline app/api/routes/order_status.py]: No need for this line

File: app/schemas/order_schema.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    status: str
+
+class OrderResponse(BaseModel):
+    id: UUID
+    user_id: Optional[UUID]
+    status: str
+    total_price: Decimal
+    created_at: datetime
+    updated_at: Optional[datetime] = None
+    products: List[dict]
```
Comment (Mohammed-samaheen) [inline app/schemas/order_schema.py]: List[OrderProduct]

File: app/schemas/order_status_schema.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    name: str
+
+class OrderStatusUpdate(BaseModel):
+    name: str
+
+class OrderStatusResponse(BaseModel):
+    id: UUID
+    name: str
+    created_at: datetime
+    updated_at: datetime
```
Comment (Mohammed-samaheen) [inline app/schemas/order_status_schema.py]: this can be null

File: app/services/order_service.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        for item in order_data.products:
+            product = self.session.get(Product, item.product_id)
+            if not product or not product.is_available:
+                raise HTTPException(status_code=404, detail="Product not found or unavailable")
+            if item.quantity > product.stock:
+                raise HTTPException(status_code=400, detail="Not enough stock for the product")
+
+            total_price += product.price * item.quantity
+            order_products_data.append(OrderProduct(product_id=product.id, quantity=item.quantity))
```
Comment (Mohammed-samaheen) [inline app/services/order_service.py:31]: You haven't deducted the product stock based on the order product quantity.

File: app/services/order_service.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+        pending_status = self.session.exec(select(OrderStatus).where(OrderStatus.name == "pending")).first()
+        new_order = Order(user_id=user_id, status_id=pending_status.id, total_price=total_price)
+        
+        self.session.add(new_order)
+        self.session.commit()
+        self.session.refresh(new_order)
+
+        for op in order_products_data:
```
Comment (Mohammed-samaheen) [inline app/services/order_service.py:40]: Just to let you know, we can add order products directly. Check this [example](https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/create-and-update-relationships/#create-a-team-with-heroes) **[no need to update the code].**

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

**Review by Mohammed-samaheen — APPROVED** [review]

**Pull request merged**